File: python-programming/examples/flask/2/echo.py
from flask import Flask, request, render_template
app = Flask(__name__)
 
@app.route("/")
def hello():
    return render_template('echo.html')
 
@app.route("/echo", methods=['POST'])
def echo(): 
    return render_template('echo.html', text=request.form['text'])
 
 
if __name__ == "__main__":
    app.run(debug=True)

File: books/mojolicious/src/examples/lite/echo.pl
use Mojolicious::Lite;



get '/' => { text => 'Hello World' };

get '/echo' => { text => q{
    <form method="POST">
    <input name="q">
    <input type="submit" value="Echo">
    </form>
}};

post '/echo' => sub {
    my $c = shift;
    $c->render( text => $c->param('q') );
};
 
app->secrets(['My very secret passphrase.']);

app->start;

